NTumbleBit/Proof/PoupardStern/Implementation/PoupardStern.cs

```diff
@@ -160,17 +160,23 @@ public static bool VerifyPoupardStern(this RsaKeyParameters pubKey, PoupardStern
             rPrime = y.Subtract(Modulus.Multiply(w));
 
             // Verifying x values
+            Dictionary<int, bool> numbers = new Dictionary<int, bool>();
             for (int i = 0; i < BigK; i++)
+                numbers.Add(i, true);
+
+            Parallel.ForEach(numbers, (n) =>
```

Comment:
`Parallel.ForEach(Enumerable.Range(0, numbers.Length), (n) =>`

Reply:
Changed to used `Enumerable.Range`